Classes/ViewHelpers/AbstractPDFViewHelper.php

```diff
@@ -90,13 +91,19 @@ public function injectConversionService(ConversionService $conversionService): v
     }
 
     /**
-     * @throws Exception
+     * @throws \Exception
      */
-    public function initializeObject(): void
+    public function setRenderingContext(RenderingContextInterface $renderingContext): void
     {
-        $this->settings = $this->configurationManager->getConfiguration(ConfigurationManagerInterface::CONFIGURATION_TYPE_SETTINGS, 'Pdfviewhelpers', 'tx_pdfviewhelpers');
+        parent::setRenderingContext($renderingContext);
 
-        if (!is_array($this->settings) || !isset($this->settings['staticTypoScriptSetupIncluded'])) {
+        $request = $this->renderingContext->getAttribute(ServerRequestInterface::class);
+        $fullTypoScript = $request->getAttribute('frontend.typoscript')->getSetupArray();
```

Comment:
Does this also work in backend context? We are currently supporting creating PDFs in the backend and would like to keep that feature.

It would probably be good to move away from the ConfigurationManager, as it's marked as internal. However I would maybe do this in a separate PR and I also just checked that VHS is also still using it for loading TypoScript: https://github.com/FluidTYPO3/vhs/blob/24c5764dcc078df1363db5c9332f9453ccfecd60/Classes/ViewHelpers/Variable/TyposcriptViewHelper.php

Reply:
If something TypoScript related has also to be working in the backend I prefer migrating that to site settings which are accessible very very fast from nearly all parts of the TYPO3 system.

Reply:
I'm not sure this kind of settings should be part of site settings, I'd rather tend to TypoScript or Page TSconfig. But even if we would go with site settings, we'd need to have a solution for this in this PR.